db/compaction/compaction.cc

```diff
@@ -268,6 +268,12 @@ Compaction::Compaction(
     max_subcompactions_ = _mutable_db_options.max_subcompactions;
   }
 
+  // for the non-bottommost levels, it tries to build files match the target
+  // file size, but not guaranteed. It could be 2x the size of the target size.
+  max_output_file_size_ = bottommost_level_ || grandparents_.empty()
```

Comment:
Does this have to take into account the new option?

Reply:
good catch, yes, it should. Updated.